test/ecto_state_machine_test.exs

```diff
@@ -95,6 +95,29 @@ defmodule EctoStateMachineTest do
     end
   end
 
+  describe "#validate_rules_change" do
+    test "valid state change", context do
+      changeset = context[:confirmed_user]
+                  |> Ecto.Changeset.change(%{rules: "admin"})
+                  |> User.validate_rules_change
+      assert changeset.valid? == true
+    end
+
+    test "invalid state change", context do
+      changeset = context[:blocked_user]
+                        |> Ecto.Changeset.change(%{rules: "admin"})
+                        |> User.validate_rules_change
+      assert changeset.valid? == false
+      assert changeset.errors == [rules: {"You can't move state from :blocked to :admin", []}]
+    end
+
+    test "changeset without state change" do
+      changeset = %User{} |> Ecto.Changeset.change(%{foo: "bar"})
+                          |> User.validate_rules_change
+      assert changeset.valid? == true
```

Comment:
This shouldn't be valid, since this `bar` state doesn't present in the `states` list.